### PR TITLE
Opt into explicit transitive content copying

### LINKED ISSUE #3864: Opt into explicit transitive content-copying in 5.0 SDK
With microsoft/msbuild#4865, MSBuild will start respecting properties that allow configurability of whether content from transitive project references is copied to the current project (right now it depends confusingly on implementation details of the projects and what's up to date). The SDK could opt into one of the new, reasonable behaviors.

### PR DESCRIPTION
**Pull request by sfoslund**
Fixes #3864

### CODE REVIEW

**Review by dsplaisted — APPROVED** [review]
How feasible is it to add a test that covers this?

**Comment by sfoslund** [comment]
@dsplaisted I'm not sure how I would go about it, do you have suggestions?

**Comment by dsplaisted** [comment]
Following the chain of linked issues, it looks like the repro steps for what fails when this isn't set would be here: https://github.com/microsoft/msbuild/issues/4677

**Comment by sfoslund** [comment]
@dsplaisted great, I've added test coverage let me know if you have any feedback.

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToReferenceAProject.cs [hunk 1]
```diff
@@ -233,5 +233,61 @@ private void AddProjectChanges(XDocument project, string childPath = null)
...
+            var testAssetB = _testAssetsManager.CreateTestProject(testProjectB);
+
+            var testProjectC = new TestProject()
+            {
+                Name = "C",
+                TargetFrameworks = targetFramework,
+                IsSdkProject = true
+            };
+            testProjectC.AdditionalProperties.Add("DisableTransitiveProjectReferences", "true");
+            testProjectC.AdditionalItems.Add("ProjectReference", Path.Combine(testAssetB.Path, testProjectB.Name, $"{testProjectB.Name}.csproj"));
```
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToReferenceAProject.cs:271]: You can create multiple `TestProjects`, add them to the `ReferencedProjects` of others as appropriate, and then create them all at once with one `CreateTestProject` call.

**Review by dsplaisted — APPROVED** [review]
Looks good.  I assume you checked that the test fails without the `MSBuildCopyContentTransitively` property set to true?

**Comment by sfoslund** [comment]
Yes, the test fails without the `MSBuildCopyContentTransitively` change.

File: src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToReferenceAProject.cs (hunk 1 above)
Comment (sfoslund) [reply src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToReferenceAProject.cs:271]: That prevents us from being able to add project changes when creating the asset, right? We need to be able to do that here to add the content with the right metadata.

Comment (dsplaisted) [reply src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToReferenceAProject.cs:271]: It makes it a bit more complicated.  There's an overload of `WithProjectChanges` that takes an `Action<string, XDocument>`.  The string you get passed will be the path to the project file, so you can write conditional logic to apply the right changes to the right file.  At least, that's how I think it's supposed to work.

**Pull request merged**